src/components/Home/Home.jsx

```diff
@@ -1,27 +1,73 @@
 import React from 'react';
-import { Redirect, useHistory } from 'react-router-dom';
+import { useState } from 'react';
+import { useForm } from '../../hooks/useForm';
+import { useFirebase } from '../../hooks/useFirebase.js';
+import { useHistory } from 'react-router-dom';
 import getToken from '../../lib/tokens';
 
 const Home = () => {
-  const userToken = localStorage.getItem('token');
+  const [values, handleInputChange, setValues] = useForm({
+    token: '',
+  });
+
+  const [msn, setMsn] = useState('');
+  const [error, setError] = useState('');
+  const [load, setLoad] = useState('');
+  const { getAll } = useFirebase();
+
+  function searching() {
+    setLoad('searching...');
+
+    var docRef = getAll().doc(values.token);
+
+    docRef
+      .get()
+      .then((doc) => {
+        if (doc.exists) {
+          setMsn('The list is found');
+          setLoad('');
+          localStorage.setItem('token', values.token);
+          history.push('/list');
+        } else {
+          setMsn('The list is not found');
+          setLoad('');
+        }
+      })
+      .catch((error) => {
+        setError(`Error getting document: ${error}`);
+      });
+  }
 
   const history = useHistory();
 
   const handleClick = () => {
     const createdToken = getToken();
-
     localStorage.setItem('token', createdToken);
-
     history.push('/list');
   };
 
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    searching();
+    setValues({
+      token: '',
+    });
+    e.target.reset();
```

Comment:
I think that it would be best that we avoid using `e.target.reset()`. The reason why this is having to be used in order to clear the form is because the inputs are not being "controlled". Right now, the <input/>'s value is controlled by the input and we are just getting told by the input what its value is. We then have to ask the input nicely to do things like clear itself via `e.target.reset()`. 

In React we typically want to be fully in control of state. We want there to be a single source of truth and we want to have undeniable ownership over that truth. Right now there are two sources of truth - the state held in the input field and the state that the input told us about, which is held in the `useForm` hook. We should be telling the input what its value is, rather than asking it politely.

In order to make the input into a controlled component, all that we need to do is pass the value associated with it into the input as a `value` prop. 
```javascript
<input value={values.token} type="text" name="token" onChange={handleChange} />
```
Voila! Now there's just one source of truth, which is held in `useForm`.  You can ditch `e.target.reset()` and know that if you change `values.token`, then the value of the input will change. :-)

If you have time, you can try and convert the other inputs in the app to controlled components. Here is a good article on controlled components from the React docs: https://es.reactjs.org/docs/forms.html#controlled-components 

 need to create a new piece of state and then pass that state into the input. Then, when

Reply:
Thank you Mike! the input is controlled now :)